[PDI-17110]: Excel Input cannot process Cognos generated Excel report

TatsianaKasiankova opened the pull request:
There are fix, unit test.

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'engine' -P '!assemblies' -Dsurefire.runOrder=alphabetical -Daudit -amd clean install
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'engine' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  engine/src/main/java/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiSheet.java
  engine/src/main/java/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxUtil.java
  engine/src/test/java/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiSheetTest.java
```

#### Unit Test Coverage

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

<details>
<summary>:warning: <b>Coverage Changes</b> (click to expand)</summary>

###### org.pentaho.di.job.entries.evaluatetablecontent.JobEntryEvalTableContent
 * Branch Change: **-2.1%** :small_red_triangle_down:
 * Complexity Change: **-2.2%** :small_red_triangle_down:
 * Instruction Change: **-0.3%** :small_red_triangle_down:

###### org.pentaho.di.job.entries.waitforsql.JobEntryWaitForSQL
 * Branch Change: **-0.8%** :small_red_triangle_down:
 * Complexity Change: **-1%** :small_red_triangle_down:
 * Instruction Change: **-0.1%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta
 * Branch Change: **-3.3%** :small_red_triangle_down:
 * Complexity Change: **-3.9%** :small_red_triangle_down:
 * Instruction Change: **-0.1%** :small_red_triangle_down:
 * Line Change: **-0.4%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.sort.SortRowsMeta
 * Instruction Change: **-0.1%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceinsert.SalesforceInsertMeta
 * Instruction Change: **-0.1%** :small_red_triangle_down:

</details>

#### License header violations

> :heavy_exclamation_mark: Copyright year is not the current year on engine/src/main/java/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxUtil.java. Found 2017. Was expecting 2018

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'engine' -P '!assemblies' -Dsurefire.runOrder=alphabetical -Daudit -amd clean install
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'engine' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  engine/src/main/java/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiSheet.java
  engine/src/main/java/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxUtil.java
  engine/src/test/java/org/pentaho/di/trans/steps/excelinput/staxpoi/StaxPoiSheetTest.java
```

#### Unit Test Coverage

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

<details>
<summary>:warning: <b>Coverage Changes</b> (click to expand)</summary>

###### org.pentaho.di.job.entries.evaluatetablecontent.JobEntryEvalTableContent
 * Branch Change: **-1.1%** :small_red_triangle_down:
 * Complexity Change: **-1.1%** :small_red_triangle_down:
 * Instruction Change: **-0.1%** :small_red_triangle_down:

###### org.pentaho.di.job.entries.ftpsget.JobEntryFTPSGet
 * Branch Change: **-0.6%** :small_red_triangle_down:
 * Complexity Change: **-0.5%** :small_red_triangle_down:
 * Instruction Change: **-0.1%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.calculator.CalculatorMetaFunction
 * Branch Change: **-8.7%** :small_red_triangle_down:
 * Complexity Change: **-3%** :small_red_triangle_down:
 * Instruction Change: **-0.7%** :small_red_triangle_down:
 * Line Change: **-0.9%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.dimensionlookup.DimensionLookupMeta
 * Branch Change: **-2.2%** :small_red_triangle_down:
 * Complexity Change: **-1.5%** :small_red_triangle_down:
 * Instruction Change: **-0.4%** :small_red_triangle_down:
 * Line Change: **-0.3%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.ifnull.IfNullMeta
 * Branch Change: **-2%** :small_red_triangle_down:
 * Complexity Change: **-1.9%** :small_red_triangle_down:
 * Instruction Change: **-0.2%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.sort.SortRowsMeta
 * Instruction Change: **-0.1%** :small_red_triangle_down:

###### org.pentaho.di.ui.repository.pur.repositoryexplorer.model.UISecurityRole
 * Branch Change: **-1.8%** :small_red_triangle_down:
 * Complexity Change: **-1.5%** :small_red_triangle_down:
 * Instruction Change: **-0.2%** :small_red_triangle_down:
 * Line Change: **-0.5%** :small_red_triangle_down:

###### org.pentaho.di.trans.steps.salesforceupsert.SalesforceUpsertMeta
 * Instruction Change: **-0.1%** :small_red_triangle_down:

</details>

TatsianaKasiankova commented:
@mbatchelor please review.
I've fixed the case with ref attribute presented as one start cell - `<dimension ref="A1"/>`.
Also I've fixed NPE occurred when we have row(s) empty before table.
Changes covered with tests.
